### ISSUE TITLE
Implement support method(s) for passing multiple files to single filter instance

### ISSUE THREAD

**Issue by stoyanK7** [checkstyle/checkstyle#12747]
## What
We need a support method for testing **two or more** files on a **single filter instance**. There is something similar for testing a two files on a single check:
https://github.com/checkstyle/checkstyle/blob/e662258faf13c3abf343563c4daeb7edec96bc9d/src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java#L240-L265

---
## Why
The request comes while working on https://github.com/checkstyle/checkstyle/pull/12151

The new filter stores file name and suppressions in the file to be more efficient. The problem is that there are numerous `pitest` mutations in this optimization because we cannot pass two files to this filter.

Similar caching behavior is implemented in `SuppressWithNearbyCommentFilter` and there is a suppressed `pitest` mutation there that can be killed with such a feature.
https://github.com/checkstyle/checkstyle/blob/e662258faf13c3abf343563c4daeb7edec96bc9d/config/pitest-suppressions/pitest-filters-suppressions.xml#L192-L199

**Comment by nrmancuso** [comment]
@stoyanK7 please send PR, we can kill mutation in https://github.com/checkstyle/checkstyle/blob/e662258faf13c3abf343563c4daeb7edec96bc9d/config/pitest-suppressions/pitest-filters-suppressions.xml#L192-L199 as proof that your new test method works.

**Comment by stoyanK7** [comment]
Uh, so I implemented the method but turns out it is not the way to kill this mutation..

A somewhat similar case(if statement with a surviving mutation) and a response from the author of `pitest`:
https://stackoverflow.com/questions/69739943/pi-test-mutation-surviving-when-trying-to-reduce-time-complexity

> As performance is not a property of code that is measured by unit testing, code that is present only to optimize performance will result in equivalent mutations that cannot be killed.
> 
> As a temporary exercise, you could also have pitest run performance tests, but this would be difficult as they normally give a quantitive result, rather than a boolean pass/fail. It would also be exceedingly slow.
> 
> As your optimization is for when n is low, it would be especially difficult to measure an observable improvement. It may be there is no actual improvement here at all. Unless you performance measurements that justify its presence, the simplest approach would be to reduce the complexity of your code by removing the if statement.

**Comment by nrmancuso** [comment]
@stoyanK7 please explain why we have this optimization (time/memory/etc)

**Comment by stoyanK7** [comment]
A reference to the file contents is stored.
https://github.com/checkstyle/checkstyle/blob/c9190ed2fbb919beafeb992bbce98912c6fa8717/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyCommentFilter.java#L308-L315
The method where the mutation happens:
https://github.com/checkstyle/checkstyle/blob/c9190ed2fbb919beafeb992bbce98912c6fa8717/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyCommentFilter.java#L326-L333
Looks like a time optimization to not call `tagSuppressions()` every time `accept()` is invoked. File contents are compared to ensure we have the tags for the current file. If file contents are not the same, the file is traversed and suppressions are saved. Tagged comments(suppressions) are stored in a list `tags`. `tagSuppressions()` is the method that runs through the whole file to add the tags(suppressions) to this list.
https://github.com/checkstyle/checkstyle/blob/c9190ed2fbb919beafeb992bbce98912c6fa8717/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyCommentFilter.java#L408-L426

The mutation would make `getFileContents()` to always return null. From:
```java
            final FileContents currentContents = event.getFileContents();

            if (getFileContents() != currentContents) {
                setFileContents(currentContents);
                tagSuppressions();
            }
```
to
```java
            final FileContents currentContents = event.getFileContents();

            if (currentContents != null) {
                setFileContents(currentContents);
                tagSuppressions();
            }
```

**Comment by rnveach** [comment]
The if statement is a "do we have a new file" check. The mutation is saying every file is considered "new".

To kill the mutation, you would need to count the number of calls to `tagSuppressions` as . It should remain at 1 as long as it continuously receives the same file with different violations.

**Comment by romani** [comment]
Please focus on functionality first, to unblock merge of filter, and pitest survival fix can be separate PR or done at the very end of implementation.
